newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/index.js

```diff
@@ -486,10 +561,69 @@ const BuildSection = React.forwardRef<Props, BuildSectionInterface>(
         )
       );
     }
+
+    // Look at projects where lastCommittedBy is not the current user, and fetch
+    // public profiles of the users that have modified them.
+    React.useEffect(
```

Comment:
I'm a bit uneasy as you said with this fairly "complex" logic, I wonder if we could simplify this to be returned by the backend in a way where it's natural to render it.

Reply:
At least let's try to move this logic out of this poor component.

Reply:
Yeah I've already improved it to avoid having to fetch contributorIds in the different versions, I think fetching the profile afterwards is fine, I'll try to make this cleaner

Reply:
I've moved this to a util function for clarity